### PR TITLE
fix: linting for app components

### PR DESCRIPTION
**Pull request by stefl**
## Description

- Linting fixes for the AppComponents directory
- Small typescript fixes to the same
- Renames ShareButttonGroup.tsx (extra t)

### CODE REVIEW

File: apps/nextjs/src/components/AppComponents/common/ShareButtonGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import { Box } from "@radix-ui/themes";
```
Comment (stefl) [inline apps/nextjs/src/components/AppComponents/common/ShareButtonGroup.tsx:1]: Typo in the filename

Comment (codeincontext) [reply apps/nextjs/src/components/AppComponents/common/ShareButtonGroup.tsx:1]: Is this a moved file?

Comment (stefl) [reply apps/nextjs/src/components/AppComponents/common/ShareButtonGroup.tsx:1]: Yes

Comment (codeincontext) [reply apps/nextjs/src/components/AppComponents/common/ShareButtonGroup.tsx:1]: It's just showing as an addition in the diff

Comment (stefl) [reply apps/nextjs/src/components/AppComponents/common/ShareButtonGroup.tsx]: I've removed and readded using git mv. Should be correct now

**Review by codeincontext — APPROVED** [review]

**Pull request merged**